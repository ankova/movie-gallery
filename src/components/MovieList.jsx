import React, {useEffect, useState, Fragment} from 'react';

const MovieList = (props) => {
    let [movieList, setMovieList] = useState([]);
    let movies = movieList;

    useEffect(() => setMovieList(props.movies))

    return (
        <div>
            <ul>
                {movies && movies.map(movie => 
                    <li key={movie.id}>
                        <h3>{movie.original_title}</h3>
                        {movie.genres.map((genre,i) => <h6 key={i}>{genre}</h6> )}
                        <img src={movie.poster_path} />
                    </li> 
                )}
            </ul>
        </div>
    );
}

export default MovieList;