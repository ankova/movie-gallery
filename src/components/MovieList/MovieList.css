/*TO DO
//Extract all fonts and colours in separate file as variables
*/
.movie-list {
  display: grid;
  grid-template-columns: auto;
  font-family: 'Open Sans', sans-serif;
  font-family: 'Kaushan Script', cursive;
  font-family: 'Saira Condensed', sans-serif; }

.movie-list ul {
  padding: 0; }

.movie-list ul li {
  box-sizing: border-box;
  max-width: 220px;
  display: inline-block;
  margin: 1em; }

.movie-list ul li img {
  width: 220px; }

.movie-list genres {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: stretch;
  align-content: space-between;
  background: black;
  height: 3em; }

.movie-list genres h6 {
  flex-flow: row;
  flex: auto; }

.movie-genres {
  width: 80%;
  margin: auto;
  margin-top: 1em; }

.movie-genres label {
  display: inline-flex;
  margin: 0.2em 0.5em; }
