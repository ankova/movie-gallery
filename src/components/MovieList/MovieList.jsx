import React, {useEffect, useState} from 'react';

import './MovieList.css';

const MovieList = (props) => {
    let [movieList, setMovieList] = useState([]);
    let movies = movieList;

    useEffect(() => setMovieList(props.movies))

    return (
        <div className="movie-list">
            <ul>
                {movies && movies.map(movie => 
                    <li key={movie.id}>
                        <div>
                            <h3>{movie.original_title}</h3>
                            <div className="genres">
                                {movie.genres.map((genre,i) => <h6 key={i}>{genre}</h6> )}
                            </div>
                            <img src={movie.poster_path} />
                        </div>
                    </li> 
                )}
            </ul>
        </div>
    );
}

export default MovieList;