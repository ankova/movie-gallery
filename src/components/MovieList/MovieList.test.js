import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import MovieList from './MovieList';

const mockProps = [
        {
            original_title: 'Title one',
            poster_path: '',
            id: 1,
            genres: ['action', 'horror']
        },
        {
            original_title: 'Title two',
            poster_path: '',
            id: 2,
            genres: ['action', 'comedy']
        }
    ];


describe('MovieList', () => {
    let component;

    it('renders component without crashing', () => {
        const tree = renderer.create(<MovieList />).toJSON;
        expect(tree).toMatchSnapshot();
    });

    beforeEach(() => component = mount(<MovieList movies={mockProps} />));
    it('should render movie list', () => {
        expect(component.find(".movie-list ul li").length).toEqual(2);
        expect(component.find(".movie-list ul li").first().find("h3").text()).toBe('Title one');
        expect(component.find(".genres h6").first().text()).toBe('action');
    });

    it('should render genres list', () => {
        expect(component.find(".genres").length).toEqual(2);
    });
});
