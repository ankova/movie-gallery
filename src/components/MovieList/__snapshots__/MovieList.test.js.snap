exports[`MovieList renders component without crashing 1`] = `[Function]`;
